Feature/516 alert UI integration

NithinL11 opened the pull request:
##  Title
**Integrate ThresholdAlertUI component to display evaluation threshold alerts (#516)**

---

##  Summary
This PR introduces a new `ThresholdAlertUI` component that integrates with the `alertThresholdService` to dynamically display an alert when the number of weekly evaluations exceeds the configured threshold.

It provides clear, real-time feedback to users (such as instructors or admins) and ensures a smooth UI experience by automatically hiding the alert once the threshold condition is no longer met.

---

##  `ThresholdAlertUI` Component Includes:

-  **Displays alert banner** when `shouldDisplayAlerts(evaluations)` returns `true`
-  **Automatically hides** the alert when evaluation counts fall below the threshold
-  **Renders the evaluation list** alongside the alert message
-  **Uses `getWeeklyThreshold()`** to fetch and show the current configured threshold value


## Testing
- Confirmed that the alert appears when evaluations exceed the threshold
- Confirmed that it disappears once evaluations are below the threshold
- Verified integration with `alertThresholdService`

NithinL11 commented:
@achow-conestogac  Please review , Thanks

arunkjoyson commented:
@achow-conestogac please review, we have implemented the usage here for alert module. thanks

achow-conestogac commented on src/Evaluation/EvaluationService/EvaluationService.ts:
```diff
@@ -43,9 +43,12 @@ export class EvaluationService implements IEvaluationService {
   loadEvaluations(): Evaluation[] {
     try {
-      const data = this.storageService.load<Evaluation[]>(EVALUATION_DATA_STORAGE_KEY);
+      const data = this.storageService.load<any[]>(EVALUATION_DATA_STORAGE_KEY);
```
why is this suddenly changed to `any`?

NithinL11 replied:
@achow-conestogac  Original code has been replaced, Please review and approve, Thanks

achow-conestogac reviewed (dismissed):
we still need the usage here.

NithinL11 commented:
@achow-conestogac , I removed that part and added original code for EvaluationService.ts kindly review and approve this, as other PR's are dependent on this.
![Image 2025-07-15 at 21 01 14_0912baa6](https://github.com/user-attachments/assets/5f8ccc1a-dda0-47ba-b55a-3db5a514b3ed)
Thank you

NithinL11 commented:
@achow-conestogac it's all resolved, please review
Thanks.
